Fixes and tweaks

Xhuis opened the pull request:
:cl: Xhuis
rscdel: Removed the global message played when Nar-Sie _begins_ to spawn (but not when it actually spawns).
tweak: Drunkenness recovery speed now increases with how drunk the imbiber is and is much quicker when the imbiber is asleep.
tweak: Suit storage units now take three seconds to enter (up from one) and have different sounds and messages for UV ray cauterization.
bugfix: Fixed some bugs with the suit storage unit, inserting mobs, and contents to seemed to duplicate themselves.
bugfix: The Summon Nar-Sie rune can now only be drawn on original station tiles and fails to invoke if scribed on the station then moved elsewhere.
/ :cl:
- Fixes #17631 
- Fixes #18102

Xhuis commented:
With the sheer amount of people that have hounded me to make this change for a very long time, you might want to consider speedmerging if it isn't too much of a problem.

phil235 commented:
If I'm reading the code correctly this means you need to drink 400u of beer to even feel slightly tispy.

Xhuis commented:
My math told me that 50u of beer would give you apx. 20 drunkenness. I haven't had time to test it in-game but it should be relatively simple.

phil235 commented:
`H.drunkenness = max((H.drunkenness + (sqrt(volume) * boozepwr * ALCOHOL_RATE)), 0)`

H.drunkenness = max((H.drunkenness + (sqrt(50) * 25 * 0.001)), 0) //new numbers for 50u beer.
sqrt(50) * 25 * 0.001 = 0.175 which is lower than 0.5 of the natural decrease, so you won't get drunk at all.

Xhuis commented:
I didn't factor in the decrease. If I just changed the passive drunkenness decrease and kept the rate as-is before this PR, would it still accomplish the effect of toning down drunkenness speed?

phil235 commented:
It would help, but I still think a system where the natural decrease varies with our drunkenness level would allow players to be tipsy or drunk more easily without risking dying from drunkenness.

Xhuis commented:
I'll see about implementing it. I'm also going to bundle a few more tweaks (mainly optimizations) and fixes into this PR.

Commit:
SSU tweaks/optimizations and alcohol changes

Xhuis commented:
All right. I changed the formulas to `drunkenness / 25` when the imbiber is awake and `drunkenness / 10` when they're asleep. I also added in some suit storage unit fixes, tweaks, and optimizations - mainly, preventing some runtime errors and strange object behavior.

Commit:
Adds LOOC

Xhuis commented:
I added LOOC, taken from other servers. It's essentially OOC that only functions within 7 tiles. Currently, it cannot be used while dead, and admins can see all LOOC that occurs. _I might relocate this to a different pull request, or remove it entirely if the maintainers don't want it._

ChangelingRain commented:
ATOMIZE YOUR PRS YA SHIT

Jordie0608 commented:
LOOC was proposed before and turned down each time thus far

Xhuis commented:
Oh, it was? Ok, I'll cut it out
@ChangelingRain I DON'T KNOW WHAT THAT MEANS

Commit:
Maybe another time

phil235 commented on code/game/machinery/suit_storage_unit.dm:
```diff
@@ -219,10 +226,15 @@
 			// The wires get damaged too.
```
can you fix a little thing above this? the code is calling qdel() for each item inside but it nullifies the reference beforehand so it currently doesn't qdel() anything. It should be qdel(X) and THEN X = null

Xhuis replied:
Nice. Yeah, I'll do that in a bit.

phil235 commented:
add Fixes #xxxx for the SSU fix.

Xhuis commented:
@phil235 If there's an issue report, I didn't see it. The issues I fixed here were discovered by myself and unreported.

Commit:
Update to latest master

Commit:
Nar-Sie rune tweaks and fixes

Xhuis commented:
Fixed #18102 and modified the Nar-Sie rune a bit (check PR body for details).

ChangelingRain commented on code/game/gamemodes/cult/ritual.dm:
```diff
@@ -214,6 +214,10 @@ This file contains the arcane tome files.
+			if(initial(AR.name) == "Space" || istype(Turf, /turf/open/space) || user.z != ZLEVEL_STATION)
+				user << "<span class='warning'>The rune cannot be scribed this far from the station!</span>"
+				return
 			var/confirm_final = alert(user, "This is the FINAL step to summon Nar-Sie, it is a long, painful ritual and the crew will be alerted to your presence", "Are you prepared for the final battle?", "My life for Nar-Sie!", "No")
```
You need to check again after the alert.

ChangelingRain commented on code/game/gamemodes/cult/ritual.dm:
```diff
@@ -214,6 +214,10 @@ This file contains the arcane tome files.
 				user << "<span class='cultlarge'>\"I am already here. There is no need to try to summon me now.\"</span>"
 				return
+			var/area/AR = get_area(user)
+			if(initial(AR.name) == "Space" || istype(Turf, /turf/open/space) || user.z != ZLEVEL_STATION)
```
The `istype(Turf, /turf/open/space)` should be for all runes tbh

Xhuis commented:
I think Travis has gone insane and doesn't want to check this PR. Nonetheless, I'll get those issues you outlined fixed shortly.

Commit:
Rune tweaks

Commit:
Whoops

Xhuis commented:
All right. Runes can't be scribed in space anymore, and runes for summoning Nar-Sie have to be on original station turf (other runes can be off-station).

optimumtact commented:
Conflicts

Xhuis commented:
### GOD
## FUCKING
# DAMN IT

Commit:
Conflicts I

Commit:
Conflicts II

Commit:
Conflicts III
